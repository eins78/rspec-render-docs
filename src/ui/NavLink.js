import React from 'react'
import kebabCase from 'lodash/kebabCase'
import deburr from 'lodash/deburr'
const slugify = str => kebabCase(deburr(str))

// document-internal link from text, self-links when isHeading
const NavLink = ({ text, isHeading, children }) => {
  const slug = slugify(text || children)
  return <a href={'#' + slug} id={isHeading ? slug : null}>{children}</a>
}

export default NavLink
