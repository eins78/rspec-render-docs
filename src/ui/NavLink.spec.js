/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import NavLink from './NavLink'

it('renders', () => {
  const component = renderer.create(
    <div>
      <NavLink isHeading text='Title as Heading'>
        Title as Heading
      </NavLink>
      <NavLink isHeading>
        Title as Children only
      </NavLink>
      <hr />
      <NavLink text='Set Title from Props but give Children'>
        Some other Children Content
      </NavLink>
      <NavLink>
        Title as Children only
      </NavLink>
    </div>
  )
  expect(component.toJSON()).toMatchSnapshot()
})
