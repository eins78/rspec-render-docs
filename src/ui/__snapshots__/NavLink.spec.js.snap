exports[`test renders 1`] = `
<div>
  <a
    href="#title-as-heading"
    id="title-as-heading">
    Title as Heading
  </a>
  <a
    href="#title-as-children-only"
    id="title-as-children-only">
    Title as Children only
  </a>
  <hr />
  <a
    href="#set-title-from-props-but-give-children"
    id={null}>
    Some other Children Content
  </a>
  <a
    href="#title-as-children-only"
    id={null}>
    Title as Children only
  </a>
</div>
`;
